### PR TITLE
Nlp candidate bert

### PR DESCRIPTION
**Pull request by yzhang123**

### CODE REVIEW

File: nemo/collections/nlp/losses/aggregator.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        loss = torch.zeros_like(values[0])
+        for loss_idx, loss_value in enumerate(values):
+            if self._weights is not None:
+                loss = loss.add(loss_value, alpha=self._weights[loss_idx])
+            else:
+                loss = loss.add(loss_value)
+        return loss
+
+    def save_to(self, save_path: str):
+        pass
```
Comment (ekmb) [inline nemo/collections/nlp/losses/aggregator.py:74]: save_to and restore_from for the loss module?

File: nemo/collections/nlp/losses/cross_entropy.py [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+
+
+def mask_padded_tokens(tokens, pad_id):
+    mask = tokens != pad_id
+    return mask
+
+
+# @experimental
+class CrossEntropyLoss(NeuralModule):
+    """
```
Comment (ekmb) [inline nemo/collections/nlp/losses/cross_entropy.py:34]: should it be in common/modules?

File: nemo/collections/nlp/losses/aggregator.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+from nemo.core.classes import NeuralModule, Typing, typecheck
+from nemo.core.neural_types import LabelsType, LengthsType, LogitsType, LogprobsType, LossType, MaskType, NeuralType
+from nemo.utils.decorators import experimental
+
+__all__ = ['AggregatorLoss']
+
+
+class AggregatorLoss(NeuralModule):
+    """
```
Comment (ekmb) [inline nemo/collections/nlp/losses/aggregator.py:28]: should it be in common/modules?

File: nemo/collections/nlp/losses/aggregator.py (hunk 1 above)
Comment (yzhang123) [reply nemo/collections/nlp/losses/aggregator.py:74]: its required since its a neural module

File: nemo/collections/common/losses/aggregator.py [hunk 4]
```diff
@@ -17,14 +17,14 @@
...
 from nemo.utils.decorators import experimental
 
 __all__ = ['AggregatorLoss']
 
 
-class AggregatorLoss(NeuralModule):
+class AggregatorLoss(nn.Module, Typing, Serialization):
     """
     Neural module which combines sums several losses into one.
     Args:
```
Comment (ekmb) [inline nemo/collections/common/losses/aggregator.py:30]: 'Neural module' - can we still call a neural module? 
`combines sums`

File: nemo/collections/common/losses/cross_entropy.py [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+    @property
+    def output_ports(self):
+        """Returns definitions of module output ports.
+        loss:
+            NeuralType(None)
+        """
+        return {"loss": NeuralType(elements_type=LossType())}
+
+    def __init__(self, logits_ndim=2, weight=None, reduction='mean'):
+        super().__init__(weight=weight, reduction=reduction)
```
Comment (ekmb) [inline nemo/collections/common/losses/cross_entropy.py:56]: could you please add docstrings/hints here, there was some confusion from the users about the logits_ndim

File: nemo/collections/nlp/losses/cross_entropy.py [hunk 6]
```diff
@@ -14,89 +14,62 @@
...
 
-def mask_padded_tokens(tokens, pad_id):
-    mask = tokens != pad_id
-    return mask
 
-
-# @experimental
-class CrossEntropyLoss(NeuralModule):
+@experimental
+class SmoothedCrossEntropy(torch.nn.Module):
```
Comment (ekmb) [inline nemo/collections/nlp/losses/cross_entropy.py:30]: rename the file with SmoothedCrossEntropy accordingly? Should it inherit from nemo.core.Loss? If so, can we combine 2 classes?

Comment (yzhang123) [reply nemo/collections/nlp/losses/cross_entropy.py:30]: i will inherit SmoothedCrossEntropyLoss from nemo.core.Loss, SmoothedCrossEntropy is just a pytorch nn.module

File: nemo/collections/common/losses/aggregator.py (hunk 4 above)
Comment (yzhang123) [reply nemo/collections/common/losses/aggregator.py:30]: i would not , its still a loss, it takes nemo.core.loss are inputs. so it should be that too. modules need restore and save_to which doesnt make sense here

File: nemo/collections/common/losses/cross_entropy.py (hunk 5 above)
Comment (yzhang123) [reply nemo/collections/common/losses/cross_entropy.py:56]: ok

File: nemo/collections/common/losses/aggregator.py (hunk 4 above)
Comment (ekmb) [reply nemo/collections/common/losses/aggregator.py:30]: exactly, my comment was regarding description please delete 'neural modules' and either 'combines' or 'sums'
